### PR TITLE
Redesigned regularCart and ICart

### PR DESCRIPTION
**Pull request by VijayS02**
Made ICart use `List<Selection>` as a selection object, where each food has a list of selection objects which corresponds to one order of that food. Each selection object corresponds to a matching index singleton object which contains addon information.

### CODE REVIEW

File: src/Entities/FoodTruck.java [hunk 1]
```diff
@@ -19,12 +19,47 @@
...
+        }
+    }
+
+    @Override
+    public boolean isAddonAvailable(IAddon add) {
+        return addonAvailability.getOrDefault(add, false);
+    }
+
+    @Override
+    public boolean isAddonListAvailable(List<IAddon> addons) {
```
Comment (akashilangovan) [inline src/Entities/FoodTruck.java:46]: Maybe rename to allAddonsAvailable?

File: src/Entities/FoodTruck.java [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+
+import java.util.HashMap;
+import java.util.List;
 
 public class FoodTruck implements IShop {
 
     Menu menu;
     IOrderbook orderbook;
-
+    HashMap<IAddon, Boolean> addonAvailability;
```
Comment (stephlum) [inline src/Entities/FoodTruck.java:12]: Why is there no boolean status attribute?

File: src/Entities/FoodTruck.java (hunk 1 above)
Comment (stephlum) [reply src/Entities/FoodTruck.java:46]: I agree - maybe getAllAddonsAvailable?

**Review by stephlum — APPROVED** [review]
Approve request

**Pull request merged**